services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/registry.py

```diff
@@ -46,47 +50,94 @@ async def _is_registry_reachable(registry_settings: RegistrySettings) -> None:
             raise _RegistryNotReachableError(error_message)
 
 
-async def wait_for_registry_liveness(app: FastAPI) -> None:
-    registry_settings: RegistrySettings = app.state.settings.REGISTRY_SETTINGS
+async def _is_dockerhub_reachable(registry_settings: RegistrySettings) -> None:
+    async with httpx.AsyncClient(timeout=5) as client:
+        params: dict[str, Any] = {"headers": {"Content-Type": "application/json"}}
+        if registry_settings.REGISTRY_AUTH:
+            params["auth"] = (
+                registry_settings.REGISTRY_USER,
+                registry_settings.REGISTRY_PW.get_secret_value(),
+            )
+            params["data"] = json.dumps(
+                {
+                    "username": registry_settings.REGISTRY_USER,
+                    "password": registry_settings.REGISTRY_PW.get_secret_value(),
+                }
+            )
+
+        # NOTE: uses different URL than the one specified in the configuration
+        url = "https://hub.docker.com/v2/users/login"
+
+        _logger.info("Registry test url ='%s'", url)
+        response = await client.post(url, **params)
+        reachable = (
+            response.status_code == status.HTTP_200_OK and "token" in response.json()
```

Comment:
Why to check `token`? What situation dockerhub responds 200 and is not "reachable"?